file.go

```diff
@@ -741,6 +741,12 @@ type FilePages struct {
 	skip       int64
 	dictionary Dictionary
 
+	// track the last page to prevent re-reading the same page
+	lastPageIndex int
+	lastPage      Page
+	lastPageValid bool
```

Comment:
Could lastPageValid be replaced with checks on `lastPage != nil`?  It looks like they are always set together.

Reply:
Yes that works. Good idea :+1: